(wixtoolset) Changed to AU package + Modernized

AdmiringWorm opened the pull request:
- Implemented AU for this package so we don't need to manually updated it each time.
- Embedded the software installer (not entirely sure if we also need to embedd the source, or if it's enough to just provide the link to the source repo)
- Updated the uninstall script so we don't need to change the registry key used for each update (tested with both 3.11 and 3.10)
- Updated the metadata (removed releaseNotes since I can't find the release notes anywhere anymore)

**NOTE: Description should be updated by someone more familiar with WiX than me**

@gep13 mind having a look since you're the original maintainer of this package.

AdmiringWorm commented:
appveyor build is failing because I've forgotten to take into account installing pre-releases. Need to look into that.

gep13 commented on automatic/wixtoolset/tools/chocolateyUninstall.ps1:
```diff
@@ -0,0 +1,28 @@
+  Write-Warning "$packageName has already been uninstalled by other means."
+} elseif ($key.Count -gt 1) {
+  Write-Warning "$($key.Count) matches found!"
+  Write-Warning "To prevent accidental data loss, no programs will be uninstall."
```
Minor nit-pick...

`no programs will be uninstalled.`

gep13 commented on automatic/wixtoolset/tools/chocolateyUninstall.ps1:
```diff
@@ -0,0 +1,28 @@
+} elseif ($key.Count -gt 1) {
+  Write-Warning "$($key.Count) matches found!"
+  Write-Warning "To prevent accidental data loss, no programs will be uninstall."
+  Write-Warning "Please alert the package maintainer the following keys were matched:"
```
Minor nit-pick...

`maintainer that the following keys were matched:`

gep13 commented on automatic/wixtoolset/wixtoolset.nuspec:
```diff
@@ -0,0 +1,34 @@
+    <tags>wixtoolset wix toolset msi xml admin</tags>
+    <summary>The most powerful set of tools available to create your Windows installation experience. Free and Open Source since 2004!</summary>
+    <description>
+The Windows Installer XML (WiX) is a toolset that builds Windows installation packages from XML source code. The toolset supports a command line environment that developers may integrate into their build processes to build MSI and MSM setup packages. The WiX toolset CodePlex project stores the stable releases and Mercurial repository.
```
This seems fine to me.  The only comment would be that with CodePlex shutting down, going forward, the last statement will cease to be true.

AdmiringWorm replied:
Oh right, actually I didn't edit the description at all.

AdmiringWorm replied:
btw, the description is imported from the Readme.md file when the package gets updated.

AdmiringWorm commented:
damn, had to look hard for the typos you noticed. I couldn't see them at first.

AdmiringWorm commented:
@gep13 I fixed the typos, I also updated the description in the Readme.md file (and added a comment in the nuspec file to not edit the description there)

AdmiringWorm commented:
@gep13 btw, what's the best way to update an existing pre-release package. How would we add the fix notation to that? just append `-20170421` and the end?

gep13 commented:
@AdmiringWorm hmm, not sure on that.  I "think" that might cause problems.

AdmiringWorm commented:
@gep13 oh well, I'll merge it anyhow and see if I can't figure something out later.
